android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboardJSHandler.java

```diff
@@ -205,15 +197,59 @@ public void setIsChiral(boolean isChiral) {
   }
 
   @JavascriptInterface
-  public abstract boolean dispatchKey(final int code, final int eventModifiers);
+  public boolean dispatchKey(final int code, final int eventModifiers) {
+    Handler mainLoop = new Handler(Looper.getMainLooper());
+    mainLoop.post(new Runnable() {
+      public void run() {
+        if (k == null) {
+          KMLog.LogError(TAG, "dispatchKey failed: Keyboard is null");
+          return;
+        }
+
+        if (k.subKeysWindow != null) {
+          return;
+        }
+
+        if (!isInappKMTextViewValid(k.keyboardType)) {
+          return;
+        }
+
+        InputConnection ic = KMManager.getInputConnection(k.keyboardType);
+        if (ic == null) {
+          KMLog.LogError(TAG, "dispatchKey failed: InputConnection is null");
+          return;
+        }
 
-  private void keyDownUp(int keyEventCode) {
+        k.dismissHelpBubble();
+        k.setShouldShowHelpBubble(false);
+
+        // Handle tab or enter since KMW didn't process it
+        if (KMManager.isDebugMode()) {
+          Log.d(TAG, "dispatchKey called with code: " + code + ", eventModifiers: " + eventModifiers);
+        }
+        if (code == KMScanCodeMap.scanCodeMap[KMScanCodeMap.KEY_TAB]) {
+          keyDownUp(KeyEvent.KEYCODE_TAB, eventModifiers);
+        } else if (code == KMScanCodeMap.scanCodeMap[KMScanCodeMap.KEY_ENTER]) {
+          keyDownUp(KeyEvent.KEYCODE_ENTER, eventModifiers);
+        }
+      }
+    });
+    return true;
+  }
+
+  private void keyDownUp(int keyEventCode, int eventModifiers) {
     if (k.keyboardType == KeyboardType.KEYBOARD_TYPE_INAPP) {
       KMTextView textView = (KMTextView)KMTextView.activeView;
-      textView.keyDownUp(KeyEvent.KEYCODE_ENTER);
+      if (keyEventCode == KeyEvent.KEYCODE_TAB) {
+        KeyEvent event = new KeyEvent(0, 0, 0, KeyEvent.KEYCODE_TAB, 0, eventModifiers, 0, 0, 0);
+        textView.dispatchKeyEvent(event);
```

Comment:
This is to account for the subtle different in IN_APP handling of KEYCODE_TAB 

Currently from master:
https://github.com/keymanapp/keyman/blob/610b9b22579fcf1cd17cf307d8243ee73f243f45/android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java#L2255-L2261